lxd/db/query/transaction.go

```diff
@@ -37,6 +37,33 @@ func Transaction(db *sql.DB, f func(*sql.Tx) error) error {
 	return err
 }
 
+// TransactionCtx executes the given function within a database transaction.
+func TransactionCtx(parentCtx context.Context, db *sql.DB, f func(context.Context, *sql.Tx) error) error {
```

Comment:
Can you update the `Transaction` function above to just call this function and passing `context.Background()` in, which will avoid duplicating this function's logic.

Reply:
Please can you call the `parentCtx` argument just `ctx` as per Go conventions.
You can then replace the variable internally with the timeout context.

Reply:
> Can you update the `Transaction` function above to just call this function and passing `context.Background()` in, which will avoid duplicating this function's logic.

`TransactionCtx` has a slightly different function arg: `f func(context.Context, *sql.Tx)` versus `f func(*sql.Tx)` which made me hesitant to replace the original `Transaction`. If the extra diff is fine with you I'll update `Transaction`.

Reply:
I would suggest updating the Transaction function to provide a wrapped f to TransactionCtx so that the f signature of Transaction doesn't change.

Reply:
Alright, should be good for another look-over :)